storage/metrics/tsicache.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sort"
+)
+
+const cacheSubsystem = "tsi_cache"     // sub-system associated with TSI index cache.
```

Comment:
Latest mystery: when I test locally, I don't see any metrics from this subsystem. I'm not sure if that's because of bugs in the code or because I'm not exercising the code-paths in the cache.